### ISSUE TITLE
Does This Still Work?

### ISSUE THREAD

**Issue by btotharye** [Vaelor/errbot-mattermost-backend#26]
No matter what I do I can't get this to work and I'm curious if anyone can confirm if this backend works?

Thanks

**Comment by Vaelor** [comment]
Yes, it still works

**Comment by btotharye** [comment]
Any suggestions on getting started with it?  I wrote the mattermost v4 api wrapper for python so I'm very familiar with mattermost and their API but even off a base clean install of errbot I can't seem to get this working on api v3 or v4.

It never seems to connect to the server

**Comment by Vaelor** [comment]
Well, assuming you followed the [Install instructions](https://github.com/Vaelor/errbot-mattermost-backend#installation), it normally should just work.
If you do have trouble, you could set the log level to debug in errbot's `config.py`:
`BOT_LOG_LEVEL = logging.DEBUG`

If there are any connection issues, they should appear there. If you want, you can attach your log/logfile here and I will have a look. (Remove any sensitive data before you do ;-) )

**Comment by btotharye** [comment]
thanks trying to figure out this issue atm why it won't find the mattermost plugin:
```
2017-09-08T21:20:45.265325+00:00 app[bot.1]: Traceback (most recent call last):
2017-09-08T21:20:45.265326+00:00 app[bot.1]:   File "/app/.heroku/python/bin/errbot", line 11, in <module>
2017-09-08T21:20:45.265327+00:00 app[bot.1]:     load_entry_point('errbot==5.1.2', 'console_scripts', 'errbot')()
2017-09-08T21:20:45.265328+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/cli.py", line 297, in main
2017-09-08T21:20:45.265329+00:00 app[bot.1]:     bootstrap(backend, root_logger, config, restore)
2017-09-08T21:20:45.265330+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/bootstrap.py", line 207, in bootstrap
2017-09-08T21:20:45.265331+00:00 app[bot.1]:     bot = setup_bot(bot_class, logger, config, restore)
2017-09-08T21:20:45.265332+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/bootstrap.py", line 130, in setup_bot
2017-09-08T21:20:45.265333+00:00 app[bot.1]:     backend_plug = backendpm.get_candidate(backend_name)
2017-09-08T21:20:45.265334+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/specific_plugin_manager.py", line 70, in get_candidate
2017-09-08T21:20:45.265334+00:00 app[bot.1]:     raise Exception("Plugin '%s' not found." % name)
2017-09-08T21:20:45.265336+00:00 app[bot.1]: Exception: Plugin 'Mattermost' not found.
```
https://github.com/btotharye/errbot-mattermost

**Comment by gbin** [comment]
Did you setup BOT_EXTRA_BACKEND_DIR correctly?

**Comment by btotharye** [comment]
BOT_EXTRA_BACKEND_DIR = '/app/plugins' yea that should be right this is how my slack version works and its fine

**Comment by Vaelor** [comment]
Using the same folder for `BOT_EXTRA_PLUGIN_DIR = "/app/plugins"` and
`BOT_EXTRA_BACKEND_DIR = '/app/plugins'` is causing the problem, you should use different folders if I recall correctly.

**Comment by btotharye** [comment]
Ok I thought I tried that, let me try again

**Comment by btotharye** [comment]
Doesn't seem to make a difference code is updated if you want to look at it

**Comment by btotharye** [comment]
It loaded hipchat fine as well, I wonder if its only loading the ones in the site packages path for some reason?

**Comment by gbin** [comment]
You don't have another error before the one you get?

**Comment by btotharye** [comment]
@gbin here is the full log don't see any other major errors and Hipchat loaded up fine just a min when I checked it so I think it has something to do with it for some reason not honoring the BOT_EXTRA_BACKEND_DIR for some reason, it prints it out right and shows the right location and the files are there when I check via console on heroku:

```
2017-09-08T22:26:24.343163+00:00 app[bot.1]: 2017-09-08 22:26:24,342 INFO     errbot.specific_plugin_manager storage search paths {'/app/.heroku/python/lib/python3.6/site-packages/errbot/storage'}
2017-09-08T22:26:24.346776+00:00 app[bot.1]: 2017-09-08 22:26:24,346 INFO     errbot.specific_plugin_manager Found those plugings available:
2017-09-08T22:26:24.347296+00:00 app[bot.1]: 2017-09-08 22:26:24,347 INFO     errbot.specific_plugin_manager 	     Shelf  (/app/.heroku/python/lib/python3.6/site-packages/errbot/storage/shelf.py)
2017-09-08T22:26:24.347566+00:00 app[bot.1]: 2017-09-08 22:26:24,347 INFO     errbot.specific_plugin_manager 	    Memory  (/app/.heroku/python/lib/python3.6/site-packages/errbot/storage/memory.py)
2017-09-08T22:26:24.348465+00:00 app[bot.1]: 2017-09-08 22:26:24,347 INFO     errbot.bootstrap          Found Storage plugin: 'Shelf'
2017-09-08T22:26:24.348591+00:00 app[bot.1]: Description: This is the storage plugin for the traditional shelf store for errbot.
2017-09-08T22:26:24.349110+00:00 app[bot.1]: 2017-09-08 22:26:24,348 DEBUG    errbot.specific_plugin_manager Refilter the plugins...
2017-09-08T22:26:24.358666+00:00 app[bot.1]: 2017-09-08 22:26:24,358 WARNING  yapsy                     Plugin candidate '/app/.heroku/python/lib/python3.6/site-packages/errbot/storage/memory.plug'  rejected by strategy 'SpecificBackendLocator'
2017-09-08T22:26:24.360663+00:00 app[bot.1]: 2017-09-08 22:26:24,360 DEBUG    errbot.specific_plugin_manager Load the one remaining...
2017-09-08T22:26:24.369348+00:00 app[bot.1]: 2017-09-08 22:26:24,369 DEBUG    errbot.specific_plugin_manager Class to load ShelfStoragePlugin
2017-09-08T22:26:24.369761+00:00 app[bot.1]: 2017-09-08 22:26:24,369 DEBUG    errbot.storage            Opening storage 'repomgr'
2017-09-08T22:26:24.369950+00:00 app[bot.1]: 2017-09-08 22:26:24,369 DEBUG    errbot.storage.shelf      Open shelf storage /tmp/repomgr.db
2017-09-08T22:26:24.394369+00:00 app[bot.1]: 2017-09-08 22:26:24,394 DEBUG    errbot.storage            Opening storage 'core'
2017-09-08T22:26:24.394685+00:00 app[bot.1]: 2017-09-08 22:26:24,394 DEBUG    errbot.storage.shelf      Open shelf storage /tmp/core.db
2017-09-08T22:26:24.402880+00:00 app[bot.1]: 2017-09-08 22:26:24,402 INFO     errbot.specific_plugin_manager backends search paths {'/app/.heroku/python/lib/python3.6/site-packages/errbot/backends'}
2017-09-08T22:26:24.413168+00:00 app[bot.1]: 2017-09-08 22:26:24,413 INFO     errbot.specific_plugin_manager Found those plugings available:
2017-09-08T22:26:24.413512+00:00 app[bot.1]: 2017-09-08 22:26:24,413 INFO     errbot.specific_plugin_manager 	      Text  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/text.py)
2017-09-08T22:26:24.413719+00:00 app[bot.1]: 2017-09-08 22:26:24,413 INFO     errbot.specific_plugin_manager 	   Hipchat  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/hipchat.py)
2017-09-08T22:26:24.413931+00:00 app[bot.1]: 2017-09-08 22:26:24,413 INFO     errbot.specific_plugin_manager 	   Graphic  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/graphic.py)
2017-09-08T22:26:24.414141+00:00 app[bot.1]: 2017-09-08 22:26:24,414 INFO     errbot.specific_plugin_manager 	      XMPP  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/xmpp.py)
2017-09-08T22:26:24.414374+00:00 app[bot.1]: 2017-09-08 22:26:24,414 INFO     errbot.specific_plugin_manager 	      Test  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/test.py)
2017-09-08T22:26:24.414642+00:00 app[bot.1]: 2017-09-08 22:26:24,414 INFO     errbot.specific_plugin_manager 	     Slack  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/slack.py)
2017-09-08T22:26:24.414984+00:00 app[bot.1]: 2017-09-08 22:26:24,414 INFO     errbot.specific_plugin_manager 	  Telegram  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/telegram_messenger.py)
2017-09-08T22:26:24.415251+00:00 app[bot.1]: 2017-09-08 22:26:24,415 INFO     errbot.specific_plugin_manager 	      Null  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/null.py)
2017-09-08T22:26:24.415453+00:00 app[bot.1]: 2017-09-08 22:26:24,415 INFO     errbot.specific_plugin_manager 	       IRC  (/app/.heroku/python/lib/python3.6/site-packages/errbot/backends/irc.py)
2017-09-08T22:26:24.415764+00:00 app[bot.1]: Traceback (most recent call last):
2017-09-08T22:26:24.415793+00:00 app[bot.1]:   File "/app/.heroku/python/bin/errbot", line 11, in <module>
2017-09-08T22:26:24.415976+00:00 app[bot.1]:     load_entry_point('errbot==5.1.2', 'console_scripts', 'errbot')()
2017-09-08T22:26:24.416002+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/cli.py", line 297, in main
2017-09-08T22:26:24.416306+00:00 app[bot.1]:     bootstrap(backend, root_logger, config, restore)
2017-09-08T22:26:24.416332+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/bootstrap.py", line 207, in bootstrap
2017-09-08T22:26:24.416563+00:00 app[bot.1]:     bot = setup_bot(bot_class, logger, config, restore)
2017-09-08T22:26:24.416591+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/bootstrap.py", line 130, in setup_bot
2017-09-08T22:26:24.416788+00:00 app[bot.1]:     backend_plug = backendpm.get_candidate(backend_name)
2017-09-08T22:26:24.416815+00:00 app[bot.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/errbot/specific_plugin_manager.py", line 70, in get_candidate
2017-09-08T22:26:24.416982+00:00 app[bot.1]:     raise Exception("Plugin '%s' not found." % name)
2017-09-08T22:26:24.417014+00:00 app[bot.1]: Exception: Plugin 'Mattermost' not found.
```

**Comment by btotharye** [comment]
has to be something odd in heroku cause this loads the plugin fine on my Digital Ocean server but now I have the original issue where it just says starting new connection to the chat server but goes no where and I know the info is all correct, I'm using the api v4 config.

Just stuck at starting new HTTPS connection to chat.mycroft.ai atm.

**Comment by gbin** [comment]
This is really strange I don't even see the bot trying to load the backend,
are you sure the path on the insurance is correct and the bot has access to
it?

On Sep 8, 2017 18:41, "Brian Hopkins" <notifications@github.com> wrote:

has to be something odd in heroku cause this loads the plugin fine on my
Digital Ocean server but now I have the original issue where it just says
starting new connection to the chat server but goes no where and I know the
info is all correct, I'm using the api v4 config.

Just stuck at starting new HTTPS connection to chat.mycroft.ai atm.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/Vaelor/errbot-mattermost-backend/issues/26#issuecomment-328230334>,
or mute the thread

**Comment by btotharye** [comment]
nvm I see what it is I tested with curl they changed the default port my bad as far as the connection not working issue.

**Comment by btotharye** [comment]
@gbin not sure I'll keep messing with heroku to find out it should.

**Comment by btotharye** [comment]
Sorry how do you change the port if they aren't using a port at all with their webhook?

**Comment by btotharye** [comment]
curling to ` https://chat.mycroft.ai:443/api/v4/users/login` works but your plugin appears to be trying to go to Not Found for url: https://chat.mycroft.ai:443/api/v4/teams/name/mycroft and fails on that part when it gets to the team, any idea what that is all about?

```
22:49:07 DEBUG    mattermostdriver.api      <Response [200]>
22:49:07 INFO     requests.packages.urllib3 Starting new HTTPS connection (1): chat.mycroft.ai
22:49:07 DEBUG    requests.packages.urllib3 "GET /api/v4/teams/name/mycroft HTTP/1.1" 404 172
22:49:07 ERROR    errbot.backends.base      Exception occurred in serve_once:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/errbot/backends/base.py", line 695, in serve_forever
    if self.serve_once():
  File "/root/backend/errbot-mattermost-backend/mattermost.py", line 385, in serve_once
    self.teamid = self.driver.api['teams'].get_team_by_name(name=self.team)['id']
  File "/usr/local/lib/python3.5/dist-packages/mattermostdriver/endpoints/teams.py", line 45, in get_team_by_name
    self.endpoint + '/name/' + name
  File "/usr/local/lib/python3.5/dist-packages/mattermostdriver/client.py", line 133, in get
    return self.make_request('get', endpoint, options=options, params=params).json()
  File "/usr/local/lib/python3.5/dist-packages/mattermostdriver/client.py", line 113, in make_request
    response.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 840, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://chat.mycroft.ai:443/api/v4/teams/name/mycroft
```

**Comment by btotharye** [comment]
seems like on this mattermost server I don't have needed permissions I'll setup my own and test it out, thanks for the info.

**Comment by btotharye** [comment]
works like a champ on my private mattermost, thanks!
